File: MovieLibrary/AngularApp/app/shared/movieOrder.ts
import { MovieItem } from "./movieItem";
import * as _ from 'lodash';

export class MovieOrder {
    id: number;
    orderDateTime: Date = new Date();
    orderNumber: string;
    items: Array<MovieItem> = new Array<MovieItem>();
    get total(): number {
       return _.sum(_.map(this.items, i=>i.price * i.quantity));
    };
}

File: MovieLibrary/Data/Entities/MovieOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.AccessControl;
using System.Threading.Tasks;

namespace MovieLibrary.Data.Entities
{
    public class MovieOrder
    {
        public int Id { get; set; }
        public DateTime OrderDateTime { get; set; }
        public string OrderNumber { get; set; }
        public ICollection<MovieItem> Items { get; set; }
        public SaveUser User { get; set; }
    }
}
